parser/quakec-parser.js

```diff
@@ -1,6 +1,56 @@
 const Lexer = require("./quakec-lexer").Lexer;
 const Range = require("./quakec-common").Range;
 
+/** @typedef {"binarySubtractRequiresLeadingWhitespace"} LanguageFeatures */
+
+class FeatureInfo {
+    constructor() {
+        this.binarySubtractRequiresLeadingWhitespace = false;
+    }
+}
+
+/** 
+ * Feature handler callback.
+ * @callback FeatureHandler
+ * @param {boolean} supported Whether the support check passed or not.
+ */
+
+class Feature {
+    /**
+     * Call a handler function. 
+     * 
+     * @param {LanguageFeatures} languageFeature 
+     * @param {FeatureHandler} handler 
+     */
+    static check(languageFeature, handler = null) {
+        handler = handler || (supported => {
+            if (!supported) {
+                Context.symbols[Context.symbols.length - 2].error("Unsupported language feature: " + languageFeature);
+            }
+        });
+
+        handler(Feature.supported(languageFeature));
+    }
+
+    /**
+     * Check for a specific language feature's support status.
+     * @param {LanguageFeatures} languageFeature 
+     * @return {boolean}
+     */
+    static supported(languageFeature) {
```

Comment:
Both `supported` and `unsupported` probably shouldn't be used outside of `Feature`? Should we rename them to `_supported` and `_unsupported`? I'd like to hear your thoughts.